client/src/services/bms/connection.ts

```diff
@@ -0,0 +1,63 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+import { BmsApi } from '@/services/bms/api';
+import { AuthenticationToken, BmsResponse, DataType, PersonalInformationResultData } from '@/services/bms/types';
+
+function assertLoggedIn<T>(value: T): asserts value is NonNullable<T> {
+  if (value === null) {
+    throw new Error('Not logged in');
+  }
+}
+
+interface ConnectionInfo {
+  token: AuthenticationToken;
+  customerInformation: PersonalInformationResultData;
+}
+
+class BmsConnection {
+  private connection: ConnectionInfo | null = null;
+
+  constructor() {}
+
+  async login(email: string, password: string): Promise<BmsResponse> {
+    const response = await BmsApi.login({ email: email, password: password });
+    if (!response.isError && response.data && response.data.type === DataType.Login) {
+      const infoResponse = await BmsApi.getPersonalInformation(response.data.token);
+      if (!infoResponse.isError && infoResponse.data && infoResponse.data.type === DataType.PersonalInformation) {
+        this.connection = {
+          token: response.data.token,
+          customerInformation: infoResponse.data,
+        };
+      }
+    }
+    return response;
+  }
+
+  async logout(): Promise<void> {
+    this.connection = null;
+  }
+
+  async getPersonalInformation(): Promise<PersonalInformationResultData> {
+    assertLoggedIn(this.connection);
+    return this.connection.customerInformation;
+  }
+
+  isLoggedIn(): boolean {
+    return this.connection !== null;
+  }
+}
+
+class ConnectionInstance {
+  private connection: BmsConnection | null = null;
+
+  constructor() {}
+
+  get(): BmsConnection {
+    if (!this.connection) {
+      this.connection = new BmsConnection();
+    }
+    return this.connection;
+  }
+}
+
+export const BmsConnectionInstance = new ConnectionInstance();
```

Comment:
Is there a reason to do this way instead of inject/provide or ways we already do in the project?

Reply:
Yeah, I suspect this is going to be used outside of a component. The more I look at it, the less I see why we bothered with inject/provide. It can be useful in some cases, like for the multi-org when we want to switch the instance being injected instead of having one global instance, but in that case we don't care: we just want one solo instance.